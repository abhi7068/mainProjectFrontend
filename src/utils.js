import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
