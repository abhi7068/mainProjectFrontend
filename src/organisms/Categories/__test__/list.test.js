import { render } from '../../../utils';

import List from '../ListCategories';
import axiosMock from 'axios';

import React from 'react';
jest.mock('axios');

const categories = {
  data: [
    {
      title: 'ddsd',
      _id: '3232323jdasdd',
    },
    {
      title: 'dd',
      _id: '3232323jdasdd',
    },
  ],
  status: 'SUCCEEDED',
};

it('snapshot', () => {
  axiosMock.get.mockImplementationOnce(() => Promise.resolve(categories));

  const { asFragment } = render(<List />, { initialState: { categories } });

  expect(
    asFragment(<List />, { initialState: { categories } })
  ).toMatchSnapshot();
});

it('mocking compoundidmount call', () => {
  axiosMock.get.mockImplementationOnce(() => Promise.resolve(categories));

  render(<List />, { initialState: { categories } });
  // expect(getByTestId('linkToUpdate')).toContainElement(link);
});
