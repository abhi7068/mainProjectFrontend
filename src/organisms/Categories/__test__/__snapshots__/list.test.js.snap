// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-47"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px; margin: 20% auto;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
